File: 4. Group Anagrams (49)/Group Anagrams (49).cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> a;
        vector<vector<string>> res;
        for(auto s: strs){
            string key = s;
            sort(key.begin(), key.end());
            a[key].push_back(s);
        }
                
        for(auto it = a.begin(); it!=a.end(); ++it){
            res.push_back(it->second);
        }
          
        return res;
    }
};

File: 4. Group Anagrams (49)/Group Anagrams (49).py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        grps = {}

        for string in strs:
            count = [0]*26
            
            for letter in string:
                count[ord(letter)-ord('a')]+=1
            
            key = '.'.join([str(n) for n in count])

            if key in grps:
                grps[key].append(string)
            else:
                grps[key] = [string]

        return grps.values()